src/ng-annotate-main.js

```diff
@@ -710,7 +727,7 @@ function followReference(node) {
         // {type: "VariableDeclarator", id: {type: "Identifier", name: "foo"}, init: ..}
         return parent;
     } else if (kind === "fun") {
-        assert(ptype === "FunctionDeclaration" || ptype === "FunctionExpression")
+        assert(ptype === "FunctionDeclaration" || isFunctionOrArrowFunctionExpression(ptype))
```

Comment:
I am not sure about this one, it depends on scope discussion (see my review comment in scopetools.js).